### ISSUE TITLE
"'numpy/arrayobject.h' file not found" error compiling on Mac OS X 10.10.3

### ISSUE THREAD

**Issue by moustakas** [esheldon/fitsio#49]
Hey Erin!

Trying to install the master branch of fitsio threw the following error:
```
python setup.py install
[snip]
creating build/lib.macosx-10.10-x86_64-2.7
creating build/lib.macosx-10.10-x86_64-2.7/fitsio
copying fitsio/__init__.py -> build/lib.macosx-10.10-x86_64-2.7/fitsio
copying fitsio/fitslib.py -> build/lib.macosx-10.10-x86_64-2.7/fitsio
copying fitsio/test.py -> build/lib.macosx-10.10-x86_64-2.7/fitsio
copying fitsio/util.py -> build/lib.macosx-10.10-x86_64-2.7/fitsio
running build_ext
building 'fitsio._fitsio_wrap' extension
gcc -fno-strict-aliasing -fno-common -dynamic -g -O2 -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/usr/local/include -I/usr/local/opt/sqlite/include -I/usr/local/Cellar/python/2.7.10/Frameworks/Python.framework/Versions/2.7/include/python2.7 -c fitsio/fitsio_pywrap.c -o build/temp.macosx-10.10-x86_64-2.7/fitsio/fitsio_pywrap.o -arch x86_64
fitsio/fitsio_pywrap.c:28:10: fatal error: 'numpy/arrayobject.h' file not found
#include <numpy/arrayobject.h> 
         ^
1 error generated.
error: command 'gcc' failed with exit status 1
```
I solved this by changing the following lines of setup.py from
```
ext=Extension("fitsio._fitsio_wrap", 
              sources,
              extra_objects=extra_objects,
              extra_compile_args=extra_compile_args, 
              extra_link_args=extra_link_args)
```
to
```
ext=Extension("fitsio._fitsio_wrap", 
              sources,
              extra_objects=extra_objects,
              extra_compile_args=extra_compile_args, 
              extra_link_args=extra_link_args,
              include_dirs=[cfitsio_dir,numpy.get_include()])
```
and everything built properly.  I can submit a PR if you'd like.

**Comment by esheldon** [comment]
strange, because I'm pretty sure I recently merged a pull request that added the numpy.get_include()

This sounds like some kind of difference in the local setup for the two systems (yours and the other author)

**Comment by beckermr** [comment]
This is def not in the master branch of fitsio and so needs to be added. In fact, for python 2.7.10 `inlcude_dirs` is not even a documented option for `setup`.

**Comment by esheldon** [comment]
John, I noticed you are using gcc.  Ive heard from a few people that clang should be used (I don't have a mac, so I can't test).  I wonder if that is why it works for others.

**Comment by esheldon** [comment]
(meaning the build of python for gcc is somehow different)

**Comment by beckermr** [comment]
This is not the problem.  If you look at the docs is sent, you will see that in Python 2.7.10, include_dirs is not an accepted keyword of distutils.core.setup. It needs to be passed to the extension as stated above.

**Comment by esheldon** [comment]
On 7/2/15, Matthew R. Becker <notifications@github.com> wrote:
> This is not the problem.  If you look at the docs is sent, you will see that
> in Python 2.7.10, include_dirs is not an accepted keyword of
> distutils.core.setup. It needs to be passed to the extension as stated
> above.

Is passing it to the extension also valid in older pythons? If not we
need a check on the version.




-- 
Erin Scott Sheldon
Brookhaven National Laboratory erin dot sheldon at gmail dot com

**Comment by beckermr** [comment]
I just checked and `include_dirs` should be passed to the extension for versions 2.4, 2.5, 2.6 and all 2.7.X  according to the docs. 

I don't know why fitsio has been building successfully, but here is a hint. I did a test with python 2.7.2 and it appears that distutils.core.setup ignores keywords it does not know. It gives a warning like this 
```/opt/TWWfsw/python27/lib/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'not_a_keyword_yet'
  warnings.warn(msg)
```
when I passed `not_a_keyword_yet=False` to it. 

For python 2.7.2, the build does *not* give this warning for `include_dirs` and also has the right paths in the compiler invocation! So apparently this functionality was undocumented and working for python 2.7.2, and now seems to not be working on my mac with python 2.7.10. I tried passing `include_dirs` in both places and got two copies of the include paths in the compiler invocation for 2.7.2. Interestingly, if I pass `include_dirs` in both places on my mac with python 2.7.10, I only get one copy of the include paths, but no warning that the keyword is unknown. I don't have access to a linux box with 2.7.10 in order to test that system+build out. 

My feeling is that the script should follow the docs, so that as undocumented keywords break or don't break, the script still works. Thus `include_dirs` should be moved to the extension.

It would be great and if guys could do a few similar tests with your versions of python on your machines to make sure nothing above is peculiar to my various setups or my current lack of sleep.

(Regarding the gcc vs clang question, note that gcc on OS X 10.9 and later (maybe 10.8 too, not sure...) is actually clang with a gcc frontend. See this blog post https://support.enthought.com/hc/en-us/articles/204469410-OS-X-GCC-Clang-and-Cython-in-10-9-Mavericks and this stack exchange http://stackoverflow.com/questions/19535422/os-x-10-9-gcc-links-to-clang, for example.)
